packages/kbn-apm-synthtrace/src/scenarios/traces_logs_assets.ts

```diff
@@ -0,0 +1,214 @@
+                  'container.name': `${serviceNames[0]}-${generateShortId()}`,
+                  'orchestrator.resource.id': generateShortId(),
+                  'cloud.provider': 'gcp',
+                  'cloud.region': 'eu-central-1',
+                  'cloud.availability_zone': 'eu-central-1a',
+                  'cloud.project.id': generateShortId(),
+                  'cloud.instance.id': generateShortId(),
+                  'log.file.path': `/logs/${generateLongId()}/error.txt`,
+                })
+                .timestamp(timestamp);
+            });
+        });
+
+      const logsOnly = range
+        .interval('1m')
+        .rate(1)
+        .generator((timestamp) => {
+          return Array(3)
+            .fill(0)
+            .map(() => {
+              const { message, level } = {
+                message: 'A simple log with something random <random> in the middle',
+                level: 'info',
+              };
+              const CLUSTER = {
+                clusterId: generateShortId(),
+                clusterName: 'synth-cluster-2',
+                namespace: 'production',
+              };
+
+              return log
+                .create()
+                .message(message.replace('<random>', generateShortId()))
+                .logLevel(level)
+                .service('synth-java')
+                .defaults({
+                  'trace.id': generateShortId(),
+                  'agent.name': 'nodejs',
+                  'orchestrator.cluster.name': CLUSTER.clusterName,
+                  'orchestrator.cluster.id': CLUSTER.clusterId,
+                  'orchestrator.namespace': CLUSTER.namespace,
+                  'container.name': `synth-java-${generateShortId()}`,
+                  'orchestrator.resource.id': generateShortId(),
+                  'cloud.provider': 'gcp',
+                  'cloud.region': 'eu-central-1',
+                  'cloud.availability_zone': 'eu-central-1a',
+                  'cloud.project.id': generateShortId(),
+                  'cloud.instance.id': generateShortId(),
+                  'log.file.path': `/logs/${generateLongId()}/error.txt`,
+                })
+                .timestamp(timestamp);
+            });
+        });
+
+      function* createGeneratorFromArray(arr: Array<Serializable<any>>) {
+        yield* arr;
+      }
+
+      const logsValuesArray = [...logsWithTraces, ...logsOnly];
+      const logsGen = createGeneratorFromArray(logsValuesArray);
+      const logsGenAssets = createGeneratorFromArray(logsValuesArray);
+
+      const traces = instances.flatMap((instance) => instanceSpans(instance));
+      const tracesGen = createGeneratorFromArray(traces);
+      const tracesGenAssets = createGeneratorFromArray(traces);
+
+      return [
+        withClient(
+          assetsEsClient,
+          logger.perf('generating_assets_events', () =>
+            Readable.from(Array.from(logsGenAssets).concat(Array.from(tracesGenAssets)))
+          )
+        ),
```

Comment:
Moving forward, we could encapsulate the asset events in a separate function that would also be responsible for copying the generators that will derive the asset documents, to make it easier to reuse this in other scenarios we'll eventually create in the future.